app/src/main/java/com/teamagam/gimelgimel/app/common/ExponentialBackoffStrategy.java

```diff
@@ -0,0 +1,49 @@
+package com.teamagam.gimelgimel.app.common;
+
+/**
+ * Exponential backoff strategy implementation.
+ * Time intervals grows exponentially between increases.
+ */
+public class ExponentialBackoffStrategy implements BackoffStrategy {
+
+    private final int mMultiplier;
+    private final int mMaxBackoffMillis;
+    private final int mBaseIntervalMillis;
+
+    private long mCurrentIntervalMillis;
+    private int mCurrentBackoffMillis;
+
+
+    public ExponentialBackoffStrategy(int baseIntervalMillis, int multiplier,
+                                      int maxBackoffMillis) {
+        mBaseIntervalMillis = baseIntervalMillis;
+        mMultiplier = multiplier;
+        mMaxBackoffMillis = maxBackoffMillis;
+
+        mCurrentIntervalMillis = baseIntervalMillis;
+        mCurrentBackoffMillis = 0;
+    }
+
+    @Override
+    public long getBackoffMillis() {
```

Comment:
also the name Backoff is confusing. maybe getCurrentIntervalMillis(). or postponing ?